### PR TITLE
Added sum and sum_if in reduction.py

### PR DESCRIPTION
**Pull request by prmukherj**
Added Sum and Sum If in reduction. Both server side and local changes are done. Changes are also done in function/reduction for immediate use.

### CODE REVIEW

**Comment by RebelYoung** [comment]
hello, how can I use the changed 'reduction.py' in my 'ansys-fluent-core` ? Copy the code into my file? Or pull your repo?

**Review by seanpearsonuk — APPROVED** [review]

**Comment by RebelYoung** [comment]
@prmukherj '[Added fallback for earlier version.]'
Hello, it means I should `pip uninstall ansys-fluent-core` , then `pip install ansys-fluent-core`  , and it would changed in the new module? even if the version is as same as former module? 

Another question, have you add the 'area_integral', 'mass_flow_rate_integral', 'volume_integral' in `solver/function/reduction.py`? I hope the same functions added in this file, because in the 'service/reduction.py' 'RuntimeError' in my PC has not been solved.
Thank you very much!!

**Review by hpohekar — APPROVED** [review]

**Comment by RebelYoung** [comment]
Thank you for your help.
I have `git pull origin feat/add_sum_and_sum_if_in_reduction` and renew the files.
However, in my workstation, there are two problems metioned previously.
The first one, "RuntimeError" occured in every function in `service/reduction`.
image

Another one, I hope synchronize all the functions in two different `reduction.py`.

File: src/ansys/fluent/core/solver/function/reduction.py [hunk 1]
```diff
@@ -166,18 +174,30 @@ def _eval_reduction(solver, reduction, locations, expr=None):
...
+            )
+            if len(locns) > 1
+            else 1
+        )
         try:
             numerator += val * extent
             denominator += extent
         except TypeError:
+            if type(val) == list:
+                return val
```
Comment (mkundu1) [inline src/ansys/fluent/core/solver/function/reduction.py:200]: Should we not need to compute the weighted average in this case?

File: setup.py
Comment (RebelYoung) [inline setup.py:1]: It solved #1981 'import failed' problem, and I will test the functions.

File: src/ansys/fluent/core/solver/function/reduction.py (hunk 1 above)
Comment (prmukherj) [reply src/ansys/fluent/core/solver/function/reduction.py]: @mkundu1, Done, Thank you.

**Review by seanpearsonuk — APPROVED** [review]

**Comment by RebelYoung** [comment]
@prmukherj @mkundu1 
The `reduction.mass_flow_integral` function returned wrong result.
image
image
It is a 2d case, so in z-coordinate the value must be 0. And the y-value is also wrong.

**Comment by RebelYoung** [comment]
I remember the result is right before the last  version.
I have tested many functions.
All of vector expressions computions returned wrong results.
image

**Comment by seanpearsonuk** [comment]
@prmukherj please could you reinstate the old tests. Then we can do comparative testing side by side. We can start by adding tests for the issues reported here and check them with both versions. Given the issues reported in 23.2 that we might assume present in 24.1 (?) we could adjust PyFluent to use the "local" service until issues are fixed in a later release.

image

**Comment by prmukherj** [comment]
> I remember the result is right before the last version. I have tested many functions. All of vector expressions computions returned wrong results. image

@RebelYoung, we tried with both 2D and 3D case but our results matched with server. If you can reproduce this issue, please file a bug with your case file and the script.
cc. @mkundu1, @seanpearsonuk

**Comment by RebelYoung** [comment]
> @RebelYoung, we tried with both 2D and 3D case but our results matched with server. If you can reproduce this issue, please file a bug with your case file and the script. cc. @mkundu1, @seanpearsonuk

@prmukherj I have not pull any change after "2023/9/18 13:44", can you tell me what is the case file used in your `test_reduction.py` ? And I can test your cas file again.

**Comment by prmukherj** [comment]
> > @RebelYoung, we tried with both 2D and 3D case but our results matched with server. If you can reproduce this issue, please file a bug with your case file and the script. cc. @mkundu1, @seanpearsonuk
> 
> @prmukherj I have not pull any change after "2023/9/18 13:44", can you tell me what is the case file used in your `test_reduction.py` ? And I can test your cas file again.

@RebelYoung, we have used: "Static_Mixer_main.cas.h5"

We can download it via:
>>> from ansys.fluent.core.examples import download_file
>>> download_file("Static_Mixer_main.cas.h5", "pyfluent/static_mixer")

**Comment by RebelYoung** [comment]
> > > @RebelYoung, we tried with both 2D and 3D case but our results matched with server. If you can reproduce this issue, please file a bug with your case file and the script. cc. @mkundu1, @seanpearsonuk
> > 
> > 
> > @prmukherj I have not pull any change after "2023/9/18 13:44", can you tell me what is the case file used in your `test_reduction.py` ? And I can test your cas file again.
> 
> @RebelYoung, we have used: "Static_Mixer_main.cas.h5"
> 
> We can download it via:
> 
> > > > from ansys.fluent.core.examples import download_file
> > > > download_file("Static_Mixer_main.cas.h5", "pyfluent/static_mixer")

@prmukherj 
I have tested the cas file "Static_Mixer_main.cas.h5" with "MassFlowInt(Velocity,['outlet'])"
The result is, there were different by two means for compute the value.

As the picture,
Firstly, we can get same values from two `reduction.py` now.
image

Secondly, `get_value()` and click `compute` also have same values, but differ from the function result before.
image
image

**Comment by RebelYoung** [comment]
@prmukherj In the 3d cas file, the difference is not as so big as my 2d cas.
There is a phenomenon , that is the value in axis=0 equal to every computing means.

**Comment by RebelYoung** [comment]
@prmukherj I tested a method : compute the vector components
```
reduction.mass_flow_integral(
    locations=sss.bc.pressure_outlet,
    expression='Velocity.x' # 'Velocity.y' , 'Velocity.z'  
)
# 1.021460818789119e-08  # -5.876406762619605e-09 #  1.660145151134899e-05
```
image

these results of vector components are right.

**Comment by RebelYoung** [comment]
@prmukherj 
```
reduction.mass_flow_integral(
    locations=sss.bc.pressure_outlet,
    expression='Velocity.x' # 'Velocity.y' , 'Velocity.z'  
)
# 1.021460818789119e-08  # -5.876406762619605e-09 #  1.660145151134899e-05

reduction.mass_flow_integral(
    locations=sss.bc.pressure_outlet,
    expression='Velocity' 
)
# [1.021460818789119e-08, 2.169100712635792e-09, 5.53526323759142e-06]
```

**Comment by prmukherj** [comment]
> @prmukherj
> 
> ```
> reduction.mass_flow_integral(
>     locations=sss.bc.pressure_outlet,
>     expression='Velocity.x' # 'Velocity.y' , 'Velocity.z'  
> )
> # 1.021460818789119e-08  # -5.876406762619605e-09 #  1.660145151134899e-05
> 
> reduction.mass_flow_integral(
>     locations=sss.bc.pressure_outlet,
>     expression='Velocity' 
> )
> # [1.021460818789119e-08, 2.169100712635792e-09, 5.53526323759142e-06]
> ```

@RebelYoung 
This is a snip-shot of the results I am getting:
image

**Comment by RebelYoung** [comment]
> > @prmukherj
> > ```
> > reduction.mass_flow_integral(
> >     locations=sss.bc.pressure_outlet,
> >     expression='Velocity.x' # 'Velocity.y' , 'Velocity.z'  
> > )
> > # 1.021460818789119e-08  # -5.876406762619605e-09 #  1.660145151134899e-05
> > 
> > reduction.mass_flow_integral(
> >     locations=sss.bc.pressure_outlet,
> >     expression='Velocity' 
> > )
> > # [1.021460818789119e-08, 2.169100712635792e-09, 5.53526323759142e-06]
> > ```
> 
> @RebelYoung This is a snip-shot of the results I am getting: image

ok, I will try to pull new code and create a new conda env to vertify the phenomonen.

**Comment by RebelYoung** [comment]
@prmukherj 
Thank you, I have pull the last code in this branch and vertified the results from those computing methods.
Now, there are identical. 
image
image
image
image

**Pull request merged**